NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/AspNetCoreEnvironmentTelemetryInitializer.cs

```diff
@@ -12,13 +12,13 @@
     public class AspNetCoreEnvironmentTelemetryInitializer : ITelemetryInitializer
     {
         private const string AspNetCoreEnvironmentPropertyName = "AspNetCoreEnvironment";
-        private readonly IHostingEnvironment environment;
+        private readonly IHostEnvironment environment;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="AspNetCoreEnvironmentTelemetryInitializer"/> class.
         /// </summary>
         /// <param name="environment">HostingEnvironment to provide EnvironmentName to be added to telemetry properties.</param>
-        public AspNetCoreEnvironmentTelemetryInitializer(IHostingEnvironment environment)
+        public AspNetCoreEnvironmentTelemetryInitializer(IHostEnvironment environment)
```

Comment:
this'd be considered breaking change?

Reply:
I'm not sure the best way to handle this. The former type appears to have been removed from .NET.

Error Message:
> 'IHostingEnvironment' is obsolete: 'This type is obsolete and will be removed in a future version.
The recommended alternative is Microsoft.AspNetCore.Hosting.IWebHostEnvironment.'

https://github.com/dotnet/AspNetCore/issues/7749

Reply:
See also: https://weblog.west-wind.com/posts/2020/Feb/26/Working-with-IWebHostEnvironment-and-IHostingEnvironment-in-dual-targeted-NET-Core-Projects

Reply:
i think i have a comprehensive fix for this.

Original constructor expected `IHostingEnvironment`. <-- now obsolete
I've added two new constructors;
- one expecting `IHostEnvironment`. <-- This is now the recommended approach.
- and one expecting both `IHostEnvironment` and `IHostingEnvironment`. <-- for backcompat

It appears that the framework can add BOTH to the Dependency Injection was causing the SDK initialization to fail because of "ambiguous constructors".
The constructor with both params solves this issue.